jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -41,6 +42,24 @@
     public static final String HTTP = "http";
     public static final String HTTPS = "https";
 
+    // From https://www.rfc-editor.org/rfc/rfc3986
+    private static final String UNRESERVED = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-._~";
+    private static final String SUBDELIMS = "!$&'()*+,;=";
+    private static final String REGNAME = UNRESERVED + SUBDELIMS;
+
+    // Allowed characters in https://www.rfc-editor.org/rfc/rfc3986 reg-name
+    private static final boolean[] REGNAME_ALLOWED;
+
+    static
+    {
+        REGNAME_ALLOWED = new boolean[127];
+        Arrays.fill(REGNAME_ALLOWED, false);
+        for (char c : REGNAME.toCharArray())
+        {
+            REGNAME_ALLOWED[c] = true;
+        }
+    }
+
     // Use UTF-8 as per http://www.w3.org/TR/html40/appendix/notes.html#non-ascii-chars
     public static final Charset __CHARSET = StandardCharsets.UTF_8;
```

Comment:
Make `private` and possibly remove the constant altogether.

Reply:
Cool, lemme do this in a different PR

Reply:
Actually, can't do this for Jetty 10.0.x (where this PR is).

BUT I will do it for Jetty 12.0.x

Reply:
Took care of this (along with other URIUtil cleanup) in `jetty-12.0.x` PR #8861